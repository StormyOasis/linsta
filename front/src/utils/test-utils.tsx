import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { RootState } from "../Components/Redux/redux";
import { authInitialState } from '../Components/Redux/slices/auth.slice';
import { modalInitialState } from '../Components/Redux/slices/modals.slice';
import { profileInitialState } from '../Components/Redux/slices/profile.slice';
import { miscInitialState } from '../Components/Redux/slices/misc.slice';
import { configureStore } from '@reduxjs/toolkit';

import Theme from '../Components/Themes/Theme';

export const mockInitialState = (overrides?: Partial<RootState>): RootState => ({
    auth: { ...authInitialState, ...(overrides?.auth ?? {}) },
    modal: { ...modalInitialState, ...(overrides?.modal ?? {}) },
    profile: { ...profileInitialState, ...(overrides?.profile ?? {}) },
    misc: { ...miscInitialState, ...(overrides?.misc ?? {}) },
});

type ExtendedRenderOptions = {
    store?: ReturnType<typeof configureStore>;
    preloadedState?: any;
    route?: string; // ‚Üê allow overriding the initial route
} & Omit<RenderOptions, "queries">;

const dummyAuthReducer = (state = { user: null, isLoading: false, error: null }, _action: any) => state;
const dummyModalReducer = (state = { isOpen: false }, _action: any) => state;
const dummyProfileReducer = (state = { name: '', email: '' }, _action: any) => state;
const dummyMiscReducer = (state = { someData: null }, _action: any) => state;

const renderWithStore = (
    ui: ReactElement,
    {
        preloadedState,
        store = configureStore({
            reducer: {
                auth: dummyAuthReducer,
                modal: dummyModalReducer,
                profile: dummyProfileReducer,
                misc: dummyMiscReducer,
            },
            preloadedState,
        }),
        route = "/",
        ...renderOptions
    }: ExtendedRenderOptions = {}
) => {
    const Wrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => {
        return (
            <Provider store={store}>
                <MemoryRouter initialEntries={[route]}>
                    <Theme>{children}</Theme>
                </MemoryRouter>
            </Provider>
        );
    };

    return {
        ...render(ui, { wrapper: Wrapper, ...renderOptions }),
        store,
    };
};

export * from "@testing-library/react";
export { renderWithStore };
export type { ExtendedRenderOptions };