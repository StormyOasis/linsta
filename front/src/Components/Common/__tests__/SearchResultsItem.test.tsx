import React from "react";
import { screen, fireEvent } from "@testing-library/react";
import SearchResultItem from "../SearchResultsItem";
import '@testing-library/jest-dom';
import { renderWithStore } from '../../../utils/test-utils';

// Mock dependencies
jest.mock("../ProfilePicMemo", () => (props: any) => (
    <img data-testid="profile-pic" alt={`${props.profile.userName}'s profile picture`} />
));
jest.mock("../Icon", () => ({
    HashtagSVG: (props: any) => <svg data-testid="hashtag-svg" {...props} />,
    SearchSVG: (props: any) => <svg data-testid="search-svg" {...props} />,
    XSVG: (props: any) => <svg data-testid="x-svg" {...props} />,
}));
jest.mock("../../../utils/utils", () => ({
    isHashtag: (text: string) => text.startsWith("#"),
}));
jest.mock("../StyledLink", () => (props: any) => (
    <a data-testid="styled-link" onClick={props.onClick} style={props.styleOverride}>{props.children}</a>
));

describe("SearchResultItem", () => {
    const profile = {
        userName: "alice",
        firstName: "Alice",
        lastName: "Smith",
        pfp: "/alice.jpg",
        profileId: "12345",
        userId: "4444",
    };

    it("renders a profile result with profile pic and user info", () => {
        const r = renderWithStore(<SearchResultItem item={profile} />);
        expect(screen.getByTestId("profile-pic")).toBeInTheDocument();
        expect(screen.getByText("alice")).toBeInTheDocument();
        expect(screen.getByText("Alice Smith")).toBeInTheDocument();
        // Should link to /alice
        expect(screen.getByRole("link", { name: "/alice" })).toHaveAttribute("href", "/alice");
        expect(r).toMatchSnapshot();
    });

    it("renders a hashtag result with HashtagSVG", () => {
        const r = renderWithStore(<SearchResultItem item="#reactjs" />);
        expect(screen.getByTestId("hashtag-svg")).toBeInTheDocument();
        expect(screen.getByText("#reactjs")).toBeInTheDocument();
        expect(screen.getByRole("link", { name: "/explore?q=%23reactjs" })).toHaveAttribute(
            "href",
            "/explore?q=%23reactjs"
        );
        expect(r).toMatchSnapshot();
    });

    it("renders a plain string result with SearchSVG", () => {
        const r = renderWithStore(<SearchResultItem item="hello" />);
        expect(screen.getByTestId("search-svg")).toBeInTheDocument();
        expect(screen.getByText("hello")).toBeInTheDocument();
        expect(screen.getByRole("link", { name: "/explore?q=hello" })).toHaveAttribute(
            "href",
            "/explore?q=hello"
        );
        expect(r).toMatchSnapshot();
    });

    it("calls onClick when result is clicked", () => {
        const handleClick = jest.fn();
        const r = renderWithStore(<SearchResultItem item={profile} onClick={handleClick} />);
        fireEvent.click(screen.getByRole("link", { name: "/alice" }));
        expect(handleClick).toHaveBeenCalled();
        expect(r).toMatchSnapshot();
    });

    it("renders remove button and calls onRemove when clicked", () => {
        const handleRemove = jest.fn();
        const r = renderWithStore(<SearchResultItem item={profile} onRemove={handleRemove} />);
        const removeBtn = screen.getByTestId("styled-link");
        fireEvent.click(removeBtn);
        expect(handleRemove).toHaveBeenCalled();
        expect(r).toMatchSnapshot();
    });
});