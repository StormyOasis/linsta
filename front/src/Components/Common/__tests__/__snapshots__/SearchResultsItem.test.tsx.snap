// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultItem calls onClick when result is clicked 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 escqfF ckFvxY"
      >
        <a
          aria-label="/alice"
          class="SearchResultsItem__NavLink-sc-hy9axe-1 iiYEDR"
          href="/alice"
          style="justify-content: normal;"
        >
          <div
            class="CombinedStyling__Div-sc-c3jdxk-0 SearchResultsItem__InnerNavLinkWrapper-sc-hy9axe-0 escqfF bZBWCW"
          >
            <div
              class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 CombinedStyling__FlexRowFullWidth-sc-c3jdxk-5 hEaoMW ckFvxY fjNDtr dVZbKj"
            >
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
              >
                <img
                  alt="alice's profile picture"
                  data-testid="profile-pic"
                />
              </div>
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexColumn-sc-c3jdxk-4 CombinedStyling__FlexColumnFullWidth-sc-c3jdxk-6 escqfF ckFvxY leZDxE dWuEVk"
              >
                <div
                  class="CombinedStyling__Div-sc-c3jdxk-0 fPYSAT"
                >
                  alice
                </div>
                <div
                  class="CombinedStyling__Div-sc-c3jdxk-0 iAChyy"
                >
                  Alice Smith
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 escqfF ckFvxY"
    >
      <a
        aria-label="/alice"
        class="SearchResultsItem__NavLink-sc-hy9axe-1 iiYEDR"
        href="/alice"
        style="justify-content: normal;"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 SearchResultsItem__InnerNavLinkWrapper-sc-hy9axe-0 escqfF bZBWCW"
        >
          <div
            class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 CombinedStyling__FlexRowFullWidth-sc-c3jdxk-5 hEaoMW ckFvxY fjNDtr dVZbKj"
          >
            <div
              class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
            >
              <img
                alt="alice's profile picture"
                data-testid="profile-pic"
              />
            </div>
            <div
              class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexColumn-sc-c3jdxk-4 CombinedStyling__FlexColumnFullWidth-sc-c3jdxk-6 escqfF ckFvxY leZDxE dWuEVk"
            >
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 fPYSAT"
              >
                alice
              </div>
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 iAChyy"
              >
                Alice Smith
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`SearchResultItem renders a hashtag result with HashtagSVG 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 escqfF ckFvxY"
      >
        <a
          aria-label="/explore?q=%23reactjs"
          class="SearchResultsItem__NavLink-sc-hy9axe-1 iiYEDR"
          href="/explore?q=%23reactjs"
          style="justify-content: normal;"
        >
          <div
            class="CombinedStyling__Div-sc-c3jdxk-0 SearchResultsItem__InnerNavLinkWrapper-sc-hy9axe-0 escqfF bZBWCW"
          >
            <div
              class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 CombinedStyling__FlexRowFullWidth-sc-c3jdxk-5 hEaoMW ckFvxY fjNDtr dVZbKj"
            >
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
              >
                <div
                  class="CombinedStyling__Div-sc-c3jdxk-0 SearchResultsItem__SearchIconWrapper-sc-hy9axe-2 dRVeqo iWRTpS"
                >
                  <svg
                    data-testid="hashtag-svg"
                    height="22px"
                    width="22px"
                  />
                </div>
              </div>
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexColumn-sc-c3jdxk-4 CombinedStyling__FlexColumnFullWidth-sc-c3jdxk-6 escqfF ckFvxY leZDxE dWuEVk"
              >
                <div
                  class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
                >
                  #reactjs
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 escqfF ckFvxY"
    >
      <a
        aria-label="/explore?q=%23reactjs"
        class="SearchResultsItem__NavLink-sc-hy9axe-1 iiYEDR"
        href="/explore?q=%23reactjs"
        style="justify-content: normal;"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 SearchResultsItem__InnerNavLinkWrapper-sc-hy9axe-0 escqfF bZBWCW"
        >
          <div
            class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 CombinedStyling__FlexRowFullWidth-sc-c3jdxk-5 hEaoMW ckFvxY fjNDtr dVZbKj"
          >
            <div
              class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
            >
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 SearchResultsItem__SearchIconWrapper-sc-hy9axe-2 dRVeqo iWRTpS"
              >
                <svg
                  data-testid="hashtag-svg"
                  height="22px"
                  width="22px"
                />
              </div>
            </div>
            <div
              class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexColumn-sc-c3jdxk-4 CombinedStyling__FlexColumnFullWidth-sc-c3jdxk-6 escqfF ckFvxY leZDxE dWuEVk"
            >
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
              >
                #reactjs
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`SearchResultItem renders a plain string result with SearchSVG 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 escqfF ckFvxY"
      >
        <a
          aria-label="/explore?q=hello"
          class="SearchResultsItem__NavLink-sc-hy9axe-1 iiYEDR"
          href="/explore?q=hello"
          style="justify-content: normal;"
        >
          <div
            class="CombinedStyling__Div-sc-c3jdxk-0 SearchResultsItem__InnerNavLinkWrapper-sc-hy9axe-0 escqfF bZBWCW"
          >
            <div
              class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 CombinedStyling__FlexRowFullWidth-sc-c3jdxk-5 hEaoMW ckFvxY fjNDtr dVZbKj"
            >
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
              >
                <div
                  class="CombinedStyling__Div-sc-c3jdxk-0 SearchResultsItem__SearchIconWrapper-sc-hy9axe-2 dRVeqo iWRTpS"
                >
                  <svg
                    data-testid="search-svg"
                    height="22px"
                    width="22px"
                  />
                </div>
              </div>
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexColumn-sc-c3jdxk-4 CombinedStyling__FlexColumnFullWidth-sc-c3jdxk-6 escqfF ckFvxY leZDxE dWuEVk"
              >
                <div
                  class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
                >
                  hello
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 escqfF ckFvxY"
    >
      <a
        aria-label="/explore?q=hello"
        class="SearchResultsItem__NavLink-sc-hy9axe-1 iiYEDR"
        href="/explore?q=hello"
        style="justify-content: normal;"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 SearchResultsItem__InnerNavLinkWrapper-sc-hy9axe-0 escqfF bZBWCW"
        >
          <div
            class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 CombinedStyling__FlexRowFullWidth-sc-c3jdxk-5 hEaoMW ckFvxY fjNDtr dVZbKj"
          >
            <div
              class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
            >
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 SearchResultsItem__SearchIconWrapper-sc-hy9axe-2 dRVeqo iWRTpS"
              >
                <svg
                  data-testid="search-svg"
                  height="22px"
                  width="22px"
                />
              </div>
            </div>
            <div
              class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexColumn-sc-c3jdxk-4 CombinedStyling__FlexColumnFullWidth-sc-c3jdxk-6 escqfF ckFvxY leZDxE dWuEVk"
            >
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
              >
                hello
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`SearchResultItem renders a profile result with profile pic and user info 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 escqfF ckFvxY"
      >
        <a
          aria-label="/alice"
          class="SearchResultsItem__NavLink-sc-hy9axe-1 iiYEDR"
          href="/alice"
          style="justify-content: normal;"
        >
          <div
            class="CombinedStyling__Div-sc-c3jdxk-0 SearchResultsItem__InnerNavLinkWrapper-sc-hy9axe-0 escqfF bZBWCW"
          >
            <div
              class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 CombinedStyling__FlexRowFullWidth-sc-c3jdxk-5 hEaoMW ckFvxY fjNDtr dVZbKj"
            >
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
              >
                <img
                  alt="alice's profile picture"
                  data-testid="profile-pic"
                />
              </div>
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexColumn-sc-c3jdxk-4 CombinedStyling__FlexColumnFullWidth-sc-c3jdxk-6 escqfF ckFvxY leZDxE dWuEVk"
              >
                <div
                  class="CombinedStyling__Div-sc-c3jdxk-0 fPYSAT"
                >
                  alice
                </div>
                <div
                  class="CombinedStyling__Div-sc-c3jdxk-0 iAChyy"
                >
                  Alice Smith
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 escqfF ckFvxY"
    >
      <a
        aria-label="/alice"
        class="SearchResultsItem__NavLink-sc-hy9axe-1 iiYEDR"
        href="/alice"
        style="justify-content: normal;"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 SearchResultsItem__InnerNavLinkWrapper-sc-hy9axe-0 escqfF bZBWCW"
        >
          <div
            class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 CombinedStyling__FlexRowFullWidth-sc-c3jdxk-5 hEaoMW ckFvxY fjNDtr dVZbKj"
          >
            <div
              class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
            >
              <img
                alt="alice's profile picture"
                data-testid="profile-pic"
              />
            </div>
            <div
              class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexColumn-sc-c3jdxk-4 CombinedStyling__FlexColumnFullWidth-sc-c3jdxk-6 escqfF ckFvxY leZDxE dWuEVk"
            >
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 fPYSAT"
              >
                alice
              </div>
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 iAChyy"
              >
                Alice Smith
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`SearchResultItem renders remove button and calls onRemove when clicked 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 escqfF ckFvxY"
      >
        <a
          aria-label="/alice"
          class="SearchResultsItem__NavLink-sc-hy9axe-1 iiYEDR"
          href="/alice"
          style="justify-content: normal;"
        >
          <div
            class="CombinedStyling__Div-sc-c3jdxk-0 SearchResultsItem__InnerNavLinkWrapper-sc-hy9axe-0 escqfF bZBWCW"
          >
            <div
              class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 CombinedStyling__FlexRowFullWidth-sc-c3jdxk-5 hEaoMW ckFvxY fjNDtr dVZbKj"
            >
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
              >
                <img
                  alt="alice's profile picture"
                  data-testid="profile-pic"
                />
              </div>
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexColumn-sc-c3jdxk-4 CombinedStyling__FlexColumnFullWidth-sc-c3jdxk-6 escqfF ckFvxY leZDxE dWuEVk"
              >
                <div
                  class="CombinedStyling__Div-sc-c3jdxk-0 fPYSAT"
                >
                  alice
                </div>
                <div
                  class="CombinedStyling__Div-sc-c3jdxk-0 iAChyy"
                >
                  Alice Smith
                </div>
              </div>
              <a
                data-testid="styled-link"
                style="margin-right: 5px;"
              >
                <svg
                  data-testid="x-svg"
                />
              </a>
            </div>
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 escqfF ckFvxY"
    >
      <a
        aria-label="/alice"
        class="SearchResultsItem__NavLink-sc-hy9axe-1 iiYEDR"
        href="/alice"
        style="justify-content: normal;"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 SearchResultsItem__InnerNavLinkWrapper-sc-hy9axe-0 escqfF bZBWCW"
        >
          <div
            class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexRow-sc-c3jdxk-3 CombinedStyling__FlexRowFullWidth-sc-c3jdxk-5 hEaoMW ckFvxY fjNDtr dVZbKj"
          >
            <div
              class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
            >
              <img
                alt="alice's profile picture"
                data-testid="profile-pic"
              />
            </div>
            <div
              class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 CombinedStyling__FlexColumn-sc-c3jdxk-4 CombinedStyling__FlexColumnFullWidth-sc-c3jdxk-6 escqfF ckFvxY leZDxE dWuEVk"
            >
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 fPYSAT"
              >
                alice
              </div>
              <div
                class="CombinedStyling__Div-sc-c3jdxk-0 iAChyy"
              >
                Alice Smith
              </div>
            </div>
            <a
              data-testid="styled-link"
              style="margin-right: 5px;"
            >
              <svg
                data-testid="x-svg"
              />
            </a>
          </div>
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
